python/segyio/tools.py

```diff
@@ -402,3 +406,178 @@ def resample(f, rate = None, delay = None, micro = False,
     f._samples = (np.arange(len(f.samples)) * rate) + t0
 
     return f
+
+
+def create_from_array2D(filename, data):
+    """ Create a new SEGY file from a 2D array
+
+    Create an unstructured SEGY file from a 2-dimentional array with defaulted
+    headers. The tracecount and samplecount are inferred from the size of the
+    array. All traceheader fields are defaulted to zero. The file have a
+    defaulted textual header and binary header. The format is defaulted
+    to IBM FLOAT 4 BYTE, and iline, xline to 189 and 193, respectively.
+
+    The 2 array is interpreted as::
+
+                       samples
+                 --------------------
+        trace 0 | s0 | s1 | ... | sn |
+                 --------------------
+        trace 1 | s0 | s1 | ... | sn |
+                 --------------------
+                          .
+                          .
+                 --------------------
+        trace n | s0 | s1 | ... | sn |
+                 --------------------
+
+    traces = [0, len(axis(0)]
+    samples = [0, len(axis(1)]
+
+    Parameters
+    ----------
+
+    filename : str
+        Path to file to create
+    data : 2-dimensional array_like
+
+    Notes
+    -----
+
+    .. version added 1.8
+
+    Examples
+    --------
+
+    Create a file from a 2D array, open it and read a trace:
+
+    >>> segyio.tools.creat_from_array(path, array2d)
+    >>> segyio.open(path, mode, strict=False) as f:
+    ...     tr = f.trace[0]
+    """
+    data = np.asarray(data)
+    dimentions = len(data.shape)
+
+    if dimentions != 2:
+        problem = "Invalid array dimentions"
+        solution = "The array must have dimentions 2 or 3"
+        raise ValueError("{},{}".format(problem, solution))
+
+    tracecount = np.size(data, 0)
+    samplecount = np.size(data, 1)
+
+    spec = segyio.spec()
+    spec.format = SegySampleFormat.IBM_FLOAT_4_BYTE
+    spec.ilines = [1]
+    spec.xlines = list(range(tracecount))
+    spec.samples = list(range(samplecount))
+    spec.sorting = TraceSortingFormat.INLINE_SORTING
+
+    with segyio.create(filename, spec) as f:
+        f.iline[1] = data
+        for tr in range(f.tracecount):
+            f.header[tr] = { TraceField.TraceNumber           : tr,
+                             TraceField.CDP_TRACE             : tr,
+                             TraceField.offset                : 1,
+                             TraceField.INLINE_3D             : 1,
+                             TraceField.CROSSLINE_3D          : tr,
+                             TraceField.TRACE_SAMPLE_COUNT    : samplecount,
+                             TraceField.TRACE_SAMPLE_INTERVAL : 4000
+                           }
+        f.bin.update(
+            tsort=2,
+            rev=1
```

Comment:
I'm a bit uncomfortable setting `rev=1`, because a compliant revision 1 file should have some more data set that we can't without help.